src/extensions/emulator/triggerHelper.ts

```diff
@@ -4,38 +4,93 @@ import {
 } from "../../emulator/functionsEmulatorShared";
 import { EmulatorLogger } from "../../emulator/emulatorLogger";
 import { Emulators } from "../../emulator/types";
-import { Resource } from "../../extensions/types";
+import {
+  Resource,
+  FUNCTIONS_RESOURCE_TYPE,
+  FUNCTIONS_V2_RESOURCE_TYPE,
+} from "../../extensions/types";
+import * as backend from "../../deploy/functions/backend";
 import * as proto from "../../gcp/proto";
+import { FirebaseError } from "../../error";
 
 /**
  * Convert a Resource into a ParsedTriggerDefinition
  */
 export function functionResourceToEmulatedTriggerDefintion(
   resource: Resource
 ): ParsedTriggerDefinition {
-  const etd: ParsedTriggerDefinition = {
-    name: resource.name,
-    entryPoint: resource.name,
-    platform: "gcfv1",
-  };
-  const properties = resource.properties || {};
-  proto.convertIfPresent(etd, properties, "timeoutSeconds", "timeout", proto.secondsFromDuration);
-  proto.convertIfPresent(etd, properties, "regions", "location", (str: string) => [str]);
-  proto.copyIfPresent(etd, properties, "availableMemoryMb");
-  if (properties.httpsTrigger) {
-    etd.httpsTrigger = properties.httpsTrigger;
+  const resourceType = resource.type;
```

Comment:
Its a little awkward to define this here and then switch between resource.type and resourceType throughout the code. Prefer getting rid of this and just using resource.type

Reply:
I had to define this var... basically if resource.type is not v1 AND not v2 then typescript infers the type of resource as `never`, so I can't use `resource` after the second if...